apps/meteor/app/livechat/server/statistics/LivechatAgentActivityMonitor.ts

```diff
@@ -52,38 +52,32 @@ export class LivechatAgentActivityMonitor {
 		// TODO use service event socket.connected instead
 		Meteor.onConnection((connection: unknown) => this._handleMeteorConnection(connection as ISocketConnection));
 		callbacks.add('livechat.agentStatusChanged', this._handleAgentStatusChanged);
-		callbacks.add('livechat.setUserStatusLivechat', async (...args) => {
+		callbacks.add('livechat.setUserStatusLivechat', (...args) => {
 			return this._handleUserStatusLivechatChanged(...args);
 		});
 		this._started = true;
 	}
 
 	async _startMonitoring(): Promise<void> {
-		await this.scheduler.add(this._name, '0 0 * * *', async () => this._updateActiveSessions());
+		await this.scheduler.add(this._name, '0 0 * * *', () => this._updateActiveSessions());
 	}
 
 	async _updateActiveSessions(): Promise<void> {
-		const openLivechatAgentSessions = await LivechatAgentActivity.findOpenSessions();
-		if (!(await openLivechatAgentSessions.count())) {
-			return;
-		}
+		const openLivechatAgentSessions = LivechatAgentActivity.findOpenSessions();
+
 		const today = moment(new Date());
 		const startedAt = new Date(today.year(), today.month(), today.date());
 		for await (const session of openLivechatAgentSessions) {
 			const startDate = moment(session.lastStartedAt);
 			const stoppedAt = new Date(startDate.year(), startDate.month(), startDate.date(), 23, 59, 59);
 			const data = { ...formatDate(startDate.toDate()), agentId: session.agentId };
 			const availableTime = moment(stoppedAt).diff(moment(new Date(session.lastStartedAt)), 'seconds');
-			await LivechatAgentActivity.updateLastStoppedAt({
-				...data,
-				availableTime,
-				lastStoppedAt: stoppedAt,
-			});
-			await LivechatAgentActivity.updateServiceHistory({
-				...data,
-				serviceHistory: { startedAt: session.lastStartedAt, stoppedAt },
-			});
-			await this._createOrUpdateSession(session.agentId, startedAt);
+
+			await Promise.all([
+				LivechatAgentActivity.updateLastStoppedAt({ ...data, availableTime, lastStoppedAt: stoppedAt }),
+				LivechatAgentActivity.updateServiceHistory({ ...data, serviceHistory: { startedAt: session.lastStartedAt, stoppedAt } }),
+				this._createOrUpdateSession(session.agentId, startedAt),
+			]);
```

Comment:
I think this is changing the behavior of the flow. Now, the previously created `session` in the last instruction could have the last `stoppedAt` and `serviceHistory` updated depending on time execution. I'd say if that's the expected behavior here, lets create another PR for this change.